lib/modules/ai_bot/personas/persona.rb

```diff
@@ -0,0 +1,119 @@
+#frozen_string_literal: true
+
+module DiscourseAi
+  module AiBot
+    module Personas
+      def self.all
+        personas = [Personas::General, Personas::SqlHelper]
+        personas << Personas::Artist if SiteSetting.ai_stability_api_key.present?
+        personas
+      end
+
+      class Persona
+        def self.name
+          I18n.t("discourse_ai.ai_bot.personas.#{to_s.demodulize.underscore}.name")
+        end
+
+        def self.description
+          I18n.t("discourse_ai.ai_bot.personas.#{to_s.demodulize.underscore}.description")
+        end
+
+        def commands
+          []
+        end
+
+        def render_commands(render_function_instructions:)
+          result = +""
+          if render_function_instructions
+            result << "\n"
+            result << function_list.system_prompt
+            result << "\n"
+          end
+          result << available_commands.map(&:custom_system_message).compact.join("\n")
+          result
+        end
+
+        def render_system_prompt(topic: nil, render_function_instructions: true)
+          substitutions = {
+            site_url: Discourse.base_url,
+            site_title: SiteSetting.title,
+            site_description: SiteSetting.site_description,
+            time: Time.zone.now,
+            commands: render_commands(render_function_instructions: render_function_instructions),
+          }
+
+          substitutions[:participants] = topic.allowed_users.map(&:username).join(", ") if topic
+
+          system_prompt.gsub(/\{(\w+)\}/) do |match|
+            found = substitutions[match[1..-2].to_sym]
+            found.nil? ? match : found.to_s
+          end
+        end
+
+        def available_commands
+          return @available_commands if @available_commands
```

Comment:
Can't you just do this?

```ruby
def available_commands
  @available_commands ||= all_available_commands.filter { |cmd| commands.include?(cmd) }
end

Reply:
Hmmm I read the rest of the class and maybe yes it's better to keep the way it is for consistency :+1: Disregard!